src/coreclr/src/jit/codegencommon.cpp

```diff
@@ -6324,71 +6331,217 @@ void CodeGen::genZeroInitFrame(int untrLclHi, int untrLclLo, regNumber initReg,
         noway_assert(uCntBytes == 0);
 
 #elif defined(TARGET_XARCH)
-        /*
-            Generate the following code:
-
-                lea     edi, [ebp/esp-OFFS]
-                mov     ecx, <size>
-                xor     eax, eax
-                rep     stosd
-         */
-
-        noway_assert(regSet.rsRegsModified(RBM_EDI));
+        assert(compiler->getSIMDSupportLevel() >= SIMD_SSE2_Supported);
+        emitter*  emit        = GetEmitter();
+        regNumber frameReg    = genFramePointerReg();
+        regNumber zeroReg     = REG_NA;
+        int       blkSize     = untrLclHi - untrLclLo;
+        int       minSimdSize = XMM_REGSIZE_BYTES;
+
+        assert(blkSize >= 0);
+        noway_assert((blkSize % sizeof(int)) == 0);
+        assert((genRegMask(initReg) & intRegState.rsCalleeRegArgMaskLiveIn) == 0); // initReg is not a live
+// incoming argument reg
+#if defined(TARGET_AMD64)
+        // We will align on x64 so can use the aligned mov
+        instruction simdMov = INS_movaps;
+        // Aligning low we want to move up to next boundary
+        int alignedLclLo = (untrLclLo + (XMM_REGSIZE_BYTES - 1)) & -XMM_REGSIZE_BYTES;
```

Comment:
Seems like this is only correct when `untrLclLo` is positive?

Reply:
Hmm, seems ok. But the assert below is firing.

Reply:
Also getting this when running benchmarks; note to self: don't squash changes until all tests pass...
```
Stack overflow.
Repeat 207 times:
--------------------------------
   at System.Array.LastIndexOf[[System.__Canon, System.Private.CoreLib, Version=5.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]](System.__Canon[], System.__Canon, Int32, Int32)
   at System.Collections.Generic.List`1[[System.__Canon, System.Private.CoreLib, Version=5.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].LastIndexOf(System.__Canon, Int32, Int32)
   at System.SR.InternalGetResourceString(System.String)
   at System.SR.GetResourceString(System.String, System.String)
   at System.NullReferenceException..ctor()
--------------------------------
   at System.Array.LastIndexOf[[System.__Canon, System.Private.CoreLib, Version=5.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]](System.__Canon[], System.__Canon, Int32, Int32)
   at System.Collections.Generic.List`1[[System.__Canon, System.Private.CoreLib, Version=5.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].LastIndexOf(System.__Canon, Int32, Int32)
   at System.SR.InternalGetResourceString(System.String)
   at System.SR.GetResourceString(System.String, System.String)
   at System.RuntimeType.GetGenericParameterConstraints()
   at System.Type.GetRootElementType()
   at System.RuntimeType.GetGenericArguments()
   at System.RuntimeType.MakeGenericType(System.Type[])
   at System.Collections.Generic.ComparerHelpers.CreateDefaultEqualityComparer(System.Type)
   at System.Collections.Generic.EqualityComparer`1[[System.__Canon, System.Private.CoreLib, Version=5.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]]..cctor()
   at System.Collections.Generic.EqualityComparer`1[[System.__Canon, System.Private.CoreLib, Version=5.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].get_Default()
   at System.Array.LastIndexOf[[System.__Canon, System.Private.CoreLib, Version=5.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]](System.__Canon[], System.__Canon, Int32, Int32)
   at System.Collections.Generic.List`1[[System.__Canon, System.Private.CoreLib, Version=5.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].LastIndexOf(System.__Canon, Int32, Int32)
   at System.SR.InternalGetResourceString(System.String)
   at System.SR.GetResourceString(System.String, System.String)
   at System.RuntimeType.GetGenericParameterConstraints()
   at System.Type.GetRootElementType()
   at System.RuntimeType+RuntimeTypeCache.GetName(System.TypeNameKind)
   at System.RuntimeType.GetCachedName(System.TypeNameKind)
   at System.RuntimeType.get_FullName()
   at System.Reflection.MemberInfo.get_CustomAttributes()
   at System.Resources.ResourceManager..ctor(System.Type)
   at System.SR.get_ResourceManager()
   at System.SR.InternalGetResourceString(System.String)
   at System.SR.GetResourceString(System.String, System.String)
   at System.ThrowHelper.GetResourceString(System.ExceptionResource)
   at System.ThrowHelper.GetArgumentOutOfRangeException(System.ExceptionArgument, System.ExceptionResource)
   at System.ThrowHelper.ThrowArgumentOutOfRangeException(System.ExceptionArgument, System.ExceptionResource)
   at System.Text.UTF8Encoding.GetBytes(System.String, Int32, Int32, Byte[], Int32)
   at System.Diagnostics.Tracing.Statics.MetadataForString(System.String, Int32, Int32, Int32)
   at System.Diagnostics.Tracing.EventSource.InitializeProviderMetadata()
   at System.Diagnostics.Tracing.EventSource.Initialize(System.Guid, System.String, System.String[])
   at System.Diagnostics.Tracing.RuntimeEventSource..ctor()
   at System.Diagnostics.Tracing.EventPipeController.Initialize()
   at System.StartupHookProvider.ProcessStartupHooks()
ExitCode != 0
```

Reply:
git still has your original commit stream in there somewhere, just not with friendly names.

Reply:
Very strange why that assert is triggering
```
low -48 : aligned -48 : size 0
low -47 : aligned -32 : size 15
low -46 : aligned -32 : size 14
low -45 : aligned -32 : size 13
low -44 : aligned -32 : size 12
low -43 : aligned -32 : size 11
low -42 : aligned -32 : size 10
low -41 : aligned -32 : size 9
low -40 : aligned -32 : size 8
low -39 : aligned -32 : size 7
low -38 : aligned -32 : size 6
low -37 : aligned -32 : size 5
low -36 : aligned -32 : size 4
low -35 : aligned -32 : size 3
low -34 : aligned -32 : size 2
low -33 : aligned -32 : size 1
low -32 : aligned -32 : size 0
low -31 : aligned -16 : size 15
low -30 : aligned -16 : size 14
low -29 : aligned -16 : size 13
low -28 : aligned -16 : size 12
low -27 : aligned -16 : size 11
low -26 : aligned -16 : size 10
low -25 : aligned -16 : size 9
low -24 : aligned -16 : size 8
low -23 : aligned -16 : size 7
low -22 : aligned -16 : size 6
low -21 : aligned -16 : size 5
low -20 : aligned -16 : size 4
low -19 : aligned -16 : size 3
low -18 : aligned -16 : size 2
low -17 : aligned -16 : size 1
low -16 : aligned -16 : size 0
low -15 : aligned 0 : size 15
low -14 : aligned 0 : size 14
low -13 : aligned 0 : size 13
low -12 : aligned 0 : size 12
low -11 : aligned 0 : size 11
low -10 : aligned 0 : size 10
low -9 : aligned 0 : size 9
low -8 : aligned 0 : size 8
low -7 : aligned 0 : size 7
low -6 : aligned 0 : size 6
low -5 : aligned 0 : size 5
low -4 : aligned 0 : size 4
low -3 : aligned 0 : size 3
low -2 : aligned 0 : size 2
low -1 : aligned 0 : size 1
low 0 : aligned 0 : size 0
low 1 : aligned 16 : size 15
low 2 : aligned 16 : size 14
low 3 : aligned 16 : size 13
low 4 : aligned 16 : size 12
low 5 : aligned 16 : size 11
low 6 : aligned 16 : size 10
low 7 : aligned 16 : size 9
low 8 : aligned 16 : size 8
low 9 : aligned 16 : size 7
low 10 : aligned 16 : size 6
low 11 : aligned 16 : size 5
low 12 : aligned 16 : size 4
low 13 : aligned 16 : size 3
low 14 : aligned 16 : size 2
low 15 : aligned 16 : size 1
low 16 : aligned 16 : size 0
low 17 : aligned 32 : size 15
low 18 : aligned 32 : size 14
low 19 : aligned 32 : size 13
low 20 : aligned 32 : size 12
low 21 : aligned 32 : size 11
low 22 : aligned 32 : size 10
low 23 : aligned 32 : size 9
low 24 : aligned 32 : size 8
low 25 : aligned 32 : size 7
low 26 : aligned 32 : size 6
low 27 : aligned 32 : size 5
low 28 : aligned 32 : size 4
low 29 : aligned 32 : size 3
low 30 : aligned 32 : size 2
low 31 : aligned 32 : size 1
low 32 : aligned 32 : size 0
low 33 : aligned 48 : size 15
low 34 : aligned 48 : size 14
low 35 : aligned 48 : size 13
low 36 : aligned 48 : size 12
low 37 : aligned 48 : size 11
low 38 : aligned 48 : size 10
low 39 : aligned 48 : size 9
low 40 : aligned 48 : size 8
low 41 : aligned 48 : size 7
low 42 : aligned 48 : size 6
low 43 : aligned 48 : size 5
low 44 : aligned 48 : size 4
low 45 : aligned 48 : size 3
low 46 : aligned 48 : size 2
low 47 : aligned 48 : size 1
```

Reply:
Previously I also had ` && defined(FEATURE_SIMD) && ALIGN_SIMD_TYPES` which removed; now the `<= sizeof(long)` test is failing, which would suggest it is unaligned by `sizeof(int)` + `sizeof(long)`? Whereas previously it was only unaligned by a pointer size...

Except... its the same configuration that was passing before and these are compiled in defines; so it should have been running that anyway, so it shouldn't have changed anything 🤔

Reply:
D'oh...

Reply:
Fixed

Reply:
🦆 debugging to the rescue